File: backend/controllers/PerfilController.php
<?php
// backend/controllers/AuthController.php
require_once ROOT . '/backend/commons/BaseController.php';
require_once ROOT . '/backend/models/UserModel.php';
require_once ROOT . '/backend/models/MiembroModel.php';

class PerfilController extends BaseController
{
  private $userModel;
  private $miembroModel;

  public function __construct()
  {
    parent::__construct();
    $this->userModel = new UserModel();
    $this->miembroModel = new MiembroModel();
  }

  public function consultarVerificacion($contra, $idPerfil)
  {
    $verificacion = $this->miembroModel->verificarMiembro($contra, $idPerfil);

    if ($verificacion) {
      $data['title'] = 'Conceptos Perfil - Arka App';
      $this->view('concepto/listar', $data);
    } else {
      $data['validation_errors'] = "Contraseña inválida";
      $this->view('auth/seleccionarperfil', $data);
    }
  }
}

File: backend/controllers/PerfilController.go
package controllers

import (
	"arka-code/backend/commons"
	"arka-code/backend/config"
	"arka-code/backend/models"
	"html/template"
	"net/http"
)

type PerfilController struct {
	userModel *models.UserModel
}

func NewPerfilController() *PerfilController {
	return &PerfilController{
		userModel: models.NewUserModel(),
	}
}

func (c *PerfilController) SolicitarPerfiles(w http.ResponseWriter, r *http.Request) {
	familyID := commons.GetFamilyID(r)
	miembros, err := c.userModel.GetMiembrosByFamily(familyID)
	if err != nil {
		commons.SetError(r, "Error al cargar perfiles")
	}

	tmpl := template.Must(template.ParseFiles("templates/auth/seleccionar-perfil.html"))
	tmpl.Execute(w, map[string]interface{}{
		"Miembros": miembros,
	})
}

func (c *PerfilController) ConsultarVerificacion(w http.ResponseWriter, r *http.Request) {
	password := r.FormValue("password")
	miembroID := commons.StringToInt(r.FormValue("idMiembro"))

	// Verificar contraseña del usuario principal
	userID := commons.GetUserID(r)
	if !c.userModel.VerifyPassword(userID, password) {
		commons.SetError(r, "Contraseña incorrecta")
		http.Redirect(w, r, config.URLROOT+"/seleccionar-perfil", http.StatusFound)
		return
	}

	commons.SetMiembroID(w, r, miembroID)
	http.Redirect(w, r, config.URLROOT+"/concepto/gasto", http.StatusFound)
}
